src/commonMain/kotlin/com/github/quillraven/fleks/collection/entityCompare.kt

```diff
@@ -4,34 +4,27 @@ import com.github.quillraven.fleks.*
 import kotlin.math.min
 
 /**
- * Sorting of int[] logic taken from: https://github.com/karussell/fastutil/blob/master/src/it/unimi/dsi/fastutil/ints/IntArrays.java
+ * Sorting of `int[]` logic taken from: https://github.com/karussell/fastutil/blob/master/src/it/unimi/dsi/fastutil/ints/IntArrays.java
  */
-interface EntityComparator {
-    fun compare(entityA: Entity, entityB: Entity): Int
-}
+fun interface EntityComparator : Comparator<Entity>
```

Comment:
Using a typealias instead of a subinterface seems to compile and work and doesn't trigger the [Exception during IR lowering](https://github.com/Quillraven/Fleks/pull/101#issuecomment-1619861790). bug.

My preference is usually to avoid type aliases and use explicit types, but in this case it's not really important so a type alias is okay. What do you think?


```suggestion
typealias EntityComparator = Comparator<Entity>
```

Reply:
I just read the last part of the official documentation: https://kotlinlang.org/docs/fun-interfaces.html#functional-interfaces-vs-type-aliases

typealias seems to be a better fit in that specific case imo. We can go ahead with that 🙂